crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -342,14 +353,101 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
         &self,
         block: &Block,
         proof: Proof,
-        inputs: HashMap<OutPoint, TxOut>,
+        inputs: UtxoMap,
         del_hashes: Vec<sha256::Hash>,
-        acc: Stump,
+        node_time: &impl NodeTime,
     ) -> Result<(), Self::Error> {
-        T::validate_block(self, block, proof, inputs, del_hashes, acc)
+        T::validate_block(self, block, proof, inputs, del_hashes, node_time)
     }
 
     fn get_fork_point(&self, block: BlockHash) -> Result<BlockHash, Self::Error> {
         T::get_fork_point(self, block)
     }
 }
+
+pub mod utxo_data {
+    use super::*;
+
+    pub struct UtxoData {
+        /// The transaction output that created this UTXO.
+        pub txout: bitcoin::TxOut,
+        //we need this mostly to validate relative-locktime.
+        /// The height of the block that the utxo was commited.
+        pub height: u32,
+        /// The timestamp of the block that the utxo was commited.
+        pub time: u32,
+    }
+
+    impl UtxoData {
+        /// Creates a new UtxoData.
+        pub const fn new(txout: bitcoin::TxOut, height: u32, time: u32) -> Self {
+            Self {
+                txout,
+                height,
+                time,
+            }
+        }
+        /// Gets the output.
+        pub const fn get_txout(&self) -> &bitcoin::TxOut {
+            &self.txout
+        }
+        /// Gets the height that the output was commited.
+        pub const fn get_height(&self) -> u32 {
+            self.height
+        }
+        pub const fn get_time(&self) -> u32 {
+            self.time
+        }
+    }
+    /// A [`HashMap<bitcoin::OutPoint, UtxoData>`] alias.
+    ///
+    /// Use HashMap to store and fast-find all needed UTXOs related data.
+    pub type UtxoMap = HashMap<bitcoin::OutPoint, UtxoData>;
+}
+
+/// Module to delegate local-time context.
+///
+/// The consumer of `Floresta-chain` has the option to implement it or just pass any u32.
+/// 0 to disable time checks.
+pub mod nodetime {
+
+    /// Disable Constant.
+    ///
+    /// Meant to be used in cases to disable time verifications.
+    pub const DISABLE_TIME: u32 = 0;
+    /// One Hour in seconds constant.
+    pub const HOUR: u32 = 60 * 60;
+
+    /// Trait to return time-related context of the chain.
+    ///
+    /// [`get_time()`] should return a the latest [unix timestamp](https://en.wikipedia.org/wiki/Unix_time) when the consumer has time-notion.
+    ///
+    /// if the consumer does not have any time notion or MTP control, its safe to use `0u32` to disable any validations on time.
+    pub trait NodeTime {
+        /// Should return a unix timestamp or 0 to skip any time related validation.
+        fn get_time(&self) -> u32;
+    }
+    impl NodeTime for u32 {
```

Comment:
This is weird, and looks like a footgun. I think this should be a zero-sized struct like:

```rust
pub struct DisableTime;

impl NodeTime for DisableTime  {
   ...
}
```

Reply:
Ill remove the direct impl for u32.

Theres no use of it because `StdNodeTime` already does a lot of the job.